internal/proc/status.go

```diff
@@ -175,15 +193,26 @@ var readStatus = func(path string) ([]string, error) {
 }
 
 // ParseStatus parses the /proc/$pid/status file and returns a *Status.
-func ParseStatus(pid string) (*Status, error) {
-	path := fmt.Sprintf("/proc/%s/status", pid)
-	lines, err := readStatus(path)
+func ParseStatus(ctx *types.PsContext, pid string) (*Status, error) {
+	var lines []string
+	var err error
+
+	if ctx.JoinUserNS {
+		lines, err = readStatusUserNS(pid)
+	} else {
+		lines, err = readStatusDefault(pid)
+	}
+
 	if err != nil {
 		return nil, err
 	}
+	return parseStatus(pid, lines)
+}
 
+// parseStatus extracts data from lines and returns a *Status.
+func parseStatus(pid string, lines []string) (*Status, error) {
 	s := Status{}
-	errUnexpectedInput := fmt.Errorf("unexpected input from %s", path)
+	errUnexpectedInput := errors.New(fmt.Sprintf("unexpected input from /proc/%s/status", pid))
```

Comment:
That wasn't intended ... need to change back to `fmt.Errorf` to make recent golints happy.